packages/discord.js/typings/index.d.ts

```diff
@@ -4794,6 +4795,7 @@ export interface ResolvedOverwriteOptions {
 }
 
 export interface RoleData {
+  reason: string;
```

Comment:
The reason should be optional. Also, you need to change the `Role#edit()` typings

```suggestion
  reason?: string;
```

Reply:
Besides this:
- the `CreateRoleOptions` interface extends this one just to add the `reason` parameter is now obsolete 
- the `PartialRoleData` interface extends this one but should not have a `reason` parameter